bdsf/readimage.py

```diff
@@ -233,32 +233,38 @@ def gaus2pix(x, location=None, use_wcs=True):
             else:
                 return img.beam2pix(x)
 
-        def pix2gaus(x, location=None, use_wcs=True):
+        def pix2gaus(x, location=None, use_wcs=True, is_error=False):
             """ Converts Gaussian parameters in pixels to deg.
 
             x - (maj [pix], min [pix], pa [deg])
             location - specifies the location in pixels (x, y) for which
                 transform is desired
             Input beam angle should be degrees CCW from the +y axis of the image.
             The output beam angle is degrees CCW from North.
+            Set is_error = True when x gives the errors on the parameters instead of
+                the parameters themselves.
             """
             if use_wcs:
                 s1, s2, th = x
                 if s1 == 0.0 and s2 == 0.0:
                     return (0.0, 0.0, 0.0)
-                brot = self.get_rot(img, location) # rotation delta CCW (in degrees) between N and +y axis of image
 
                 th_rad = th / 180.0 * N.pi
                 bmaj = self.pixdist2angdist(img, s1, th, location=location)
                 bmin = self.pixdist2angdist(img, s2, th + 90.0, location=location)
-                bpa = th - brot
-                if bmaj < bmin:
-                    bmaj, bmin = bmin, bmaj
-                    bpa += 90.0
-                bpa = divmod(bpa, 180)[1] ### bpa lies between 0 and 180
+                bpa = th
+                if not is_error:
+                    # Adjust the PA by the rotation delta and fix cases where
+                    # major and minor axes are swapped
+                    brot = self.get_rot(img, location) # rotation delta CCW (in degrees) between N and +y axis of image
+                    bpa = th - brot
+                    if bmaj < bmin:
+                        bmaj, bmin = bmin, bmaj
+                        bpa += 90.0
+                    bpa = divmod(bpa, 180)[1] ### bpa lies between 0 and 180
```

Comment:
Is there no `else` clause, or did you forget to add it?

Reply:
No, no other processing is needed in that case.